server/controllers/article.js

```diff
@@ -67,15 +68,21 @@ class Article {
    */
   static async getArticle(request, response) {
     const articleSlug = request.params.slug;
+    const { userId } = request;
     const article = await articleRepo.getSingleArticle(articleSlug);
-
     if (!article) {
       return badHttpResponse(
         response,
         404,
         'This article was not found',
       );
     }
+
+    if (userId && userId !== article.userid) {
+      const user = await userRepo.getUserByParam('id', userId);
+      await articleRepo.addReadingStat(user, article);
```

Comment:
you should pass in the user id here. No need for another db call at this point

Reply:
Noted.